test/e2e/es_cluster_logging.go

```diff
@@ -0,0 +1,293 @@
+		Failf("Elasticsearch cluster has a bad status: %v", statusCode)
+	}
+	// Check to see if have a cluser_name field.
+	clusterName, ok := esResponse["cluster_name"]
+	if !ok {
+		Failf("No cluster_name field in Elasticsearch response: %v", esResponse)
+	}
+	if clusterName != "kubernetes_logging" {
+		Failf("Connected to wrong cluster %q (expecting kubernetes_logging)", clusterName)
+	}
+
+	// Now assume we really are talking to an Elasticsearch instance.
+	// Check the cluster health.
+	By("Checking health of Elasticsearch service.")
+	body, err = c.Get().Prefix("proxy").Resource("services").Name("elasticsearch-logging").Suffix("_cluster/health").Param("health", "pretty").DoRaw()
+	Expect(err).NotTo(HaveOccurred())
+	health, err := bodyToJSON(body)
+	Expect(err).NotTo(HaveOccurred())
+	if statusIntf, ok = health["status"]; !ok {
+		Failf("No status field found in cluster health response: %v", health)
+	}
+	status := statusIntf.(string)
+	if status != "green" && status != "yellow" {
+		Failf("Cluster health has bad status: %s", status)
+	}
+
+	// Obtain a list of nodes so we can place one synthetic logger on each node.
+	nodes, err := c.Nodes().List()
+	if err != nil {
+		Failf("Failed to list nodes: %v", err)
+	}
+	nodeCount := len(nodes.Items)
+	if nodeCount == 0 {
+		Failf("Failed to find any nodes")
+	}
+
+	// Create a unqiue root name for the resources in this test to permit
+	// parallel executions of this test.
+	name := "synthlogger-" + string(util.NewUUID())
+	// Replace '-' characters with '_' to prevent the analyzer from breaking apart names.
+	underscoreName := strings.Replace(name, "-", "_", -1)
+
+	podNames := []string{}
+	const countTo = 100
+	// Instantiate a synthetic logger pod on each node.
+	for i, node := range nodes.Items {
+		podName := fmt.Sprintf("%s-%d", name, i)
+		_, err := c.Pods(ns).Create(&api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"name": name},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    "synth-logger",
+						Image:   "ubuntu:14.04",
+						Command: []string{"bash", "-c", fmt.Sprintf("i=0; while ((i < %d)); do echo \"%d %s $i %s\"; i=$(($i+1)); done", countTo, i, underscoreName, podName)},
+					},
+				},
+				Host:          node.Name,
+				RestartPolicy: api.RestartPolicy{Never: &api.RestartPolicyNever{}},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		podNames = append(podNames, podName)
+	}
+
+	// Cleanup the pods when we are done.
+	defer func() {
+		for _, pod := range podNames {
+			if err = c.Pods(ns).Delete(pod); err != nil {
+				Logf("Failed to delete pod %s: %v", pod, err)
+			}
+		}
+	}()
+
+	// Wait for the syntehtic logging pods to finish.
+	By("Waiting for the pods to succeed.")
+	for _, pod := range podNames {
+		err = waitForPodSuccess(c, pod, "synth-logger")
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// Wait a bit for the log information to make it into Elasticsearch.
```

Comment:
Is there a way to poll ES so that we know when it's ready to be queried rather than doing a sleep here?